dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-script/src/main/java/org/apache/dolphinscheduler/plugin/alert/script/ScriptSender.java

```diff
@@ -69,6 +69,11 @@ private AlertResult executeShellScript(String title, String content) {
             alertResult.setMessage("shell script not support windows os");
             return alertResult;
         }
+        if (!scriptPath.endsWith(".sh")) {
+            alertResult.setMessage("shell script is invalid, only support .sh file");
```

Comment:
Is it possible for users to put something like `reboot` in the script?

Reply:
Yes, but it seems we can't limit user custom script content.  If add a blacklist, it is easy to miss some keys.

Reply:
In that case, we have to deprecate this plugin in the future. I think users tend to deploy a centralized alert server, which is different from worker and makes the alert server more vulnerable.